src-ext/TypeScript/ggvega/src/encodingByAes.ts

```diff
@@ -0,0 +1,154 @@
+import {contains} from './utils';
+import * as VL from './vlSpec';
+import {VLEncodingField} from './itmEncodingObject';
+import {GGEncodingKey} from './encodingNameByGeom';
+
+/**
+ * Create an empty `Encoding` according to an aesthetic-name
+ *
+ * @remarks
+ * There are different "flavors" of Vega-Lite encodings; which one is used
+ * depends on the encoding key (name), which in turn depends on the
+ * aesthetic name.
+ *
+ * In the future, maybe we could imagine an `opts` argument so that we could
+ * instantiate the encoding and set values in one step.
+ *
+ * **Called by**
+ * @see {@link itmEncodingObjectByMappingObject}
+ * @see {@link itmEncodingObjectByAesParamsObject}
+ *
+ * **Calls**
+ * @see encodingX
+ * @see encodingY
+ * @see encodingString
+ * @see encodingNumber
+ * @see encodingShape
+ * @see encodingDetail
+ *
+ * @param aesName - `string` name of ggplot2 aesthetic
+ *
+ * @returns `VL.Encoding` empty encoding
+ */
+export function encodingByAes(aesName: string): VLEncodingField {
+  // assuming that VL.ValueDefWithConditionMarkPropFieldDefNumber, etc.
+  // are all subclasses of VL.Encoding
+
+  //NOTE @wenyu: Add group` and `weight` to GG.Mapping
```

Comment:
I think you can remove this note as you have done this.

Reply:
done.